From 98dec8d10fa401aabc1da4caa571df964f6bfc13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabiano=20Fid=C3=AAncio?= <fidencio@redhat.com>
Date: Tue, 23 Sep 2014 16:07:50 +0200
Subject: [PATCH] Simplify virt_viewer_auth_libvirt_credentials() return value
 logic

As the function should return < 0 in error cases, let's explicitly
return/set the return value to -1 in error cases. Otherwise, the
function will return 0.

This patch also fixes a regression introduced by (a5ce2ed3).

https://bugzilla.redhat.com/show_bug.cgi?id=1142742
(cherry picked from commit 42ffe6f8b4efe2a1e80fc98dbded37823133b55c)
---
 src/virt-viewer.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/virt-viewer.c b/src/virt-viewer.c
index aceb474..1d14b70 100644
--- a/src/virt-viewer.c
+++ b/src/virt-viewer.c
@@ -671,7 +671,7 @@ virt_viewer_auth_libvirt_credentials(virConnectCredentialPtr cred,
     VirtViewer *app = cbdata;
     VirtViewerPrivate *priv = app->priv;
     int i;
-    int ret = -1;
+    int ret = 0;
 
     g_debug("Got libvirt credential request for %d credential(s)", ncred);
 
@@ -697,10 +697,10 @@ virt_viewer_auth_libvirt_credentials(virConnectCredentialPtr cred,
                                                                      "libvirt",
                                                                      app->priv->uri,
                                                                      username, password);
-        if (priv->auth_cancelled)
+        if (priv->auth_cancelled) {
+            ret = -1;
             goto cleanup;
-    } else {
-        ret = 0;
+        }
     }
 
     for (i = 0 ; i < ncred ; i++) {
